### PR TITLE
SSE-C: Add support in Bucket Post Policy

### LINKED ISSUE #5606: POST object should support encryption in form fields
https://docs.aws.amazon.com/AmazonS3/latest/dev/ServerSideEncryptionCustomerKeys.html

## Current Behavior
We do not support it yet.

## Possible Solution
https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPOST.html

> Server-Side Encryption Specific Request Form Fields

Name | Description | Required
-- | -- | --
x-amz-server-side-encryption-customer-algorithm | Specifies the algorithm to use to when encrypting the object.Type: StringDefault: NoneValid Value: AES256Constraints: Must be accompanied by valid x-amz-server-side-encryption-customer-key and x-amz-server-side-encryption-customer-key-MD5 fields. | Yes
x-amz-server-side-encryption-customer-key | Specifies the customer-provided base64-encoded encryption key for Amazon S3 to use in encrypting data. This value is used to store the object and then it is discarded; Amazon does not store the encryption key. The key must be appropriate for use with the algorithm specified in the x-amz-server-side-encryption-customer-algorithm header.Type: StringDefault: NoneConstraints: Must be accompanied by valid x-amz-server-side-encryption-customer-algorithm and x-amz-server-side-encryption-customer-key-MD5 fields. | Yes
x-amz-server-side-encryption-customer-key-MD5 | Specifies the base64-encoded 128-bit MD5 digest of the encryption key according to RFC 1321. Amazon S3 uses this header for a message integrity check to ensure the encryption key was transmitted without error.Type: StringDefault: NoneConstraints: Must be accompanied by valid x-amz-server-side-encryption-customer-algorithm and x-amz-server-side-encryption-customer-key fields. | Yes

## Steps to Reproduce (for bugs)
We don't support it. 

## Context
SSE-C support completion

## Your Environment
master

### PR DESCRIPTION
**Pull request by vadmeste**
## Description
Add support of server side encryption in Bucket Post Policy

## Motivation and Context
Fixes #5606 

## How Has This Been Tested?
Manually

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

File: cmd/bucket-handlers.go [hunk 1]
```diff
@@ -566,6 +566,29 @@ func (api objectAPIHandlers) PostPolicyBucketHandler(w http.ResponseWriter, r *h
 		return
 	}
 
+	if objectAPI.IsEncryptionSupported() {
+		if IsSSECustomerRequest(formValues) && !hasSuffix(object, slashSeparator) { // handle SSE-C requests
```
Comment (harshavardhana) [inline cmd/bucket-handlers.go]: you want to name this as `IsSSECustomerHeader` ?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/bucket-handlers.go (hunk 1 above)
Comment (vadmeste) [reply cmd/bucket-handlers.go]: > you want to name this as IsSSECustomerHeader ?

I renamed to hasSSECustomerHeader. What do you think ?

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

**Review by harshavardhana — APPROVED** [review]
LGTM works as expected.

**Review by aead — APPROVED** [review]

**Pull request merged**